### PR TITLE
Add circleci config to test cookiecutter

### PR DESCRIPTION
**Pull request by adamhadani**
- Add CircleCI config boilerplate
- Added GitHub CODEOWNERS defaults
- Add code to instantiate cookiecutter into Docker container and run test, lint and typehinting on that container

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      - run:
+          name: Build Base Docker
+          # install dependencies for loading ecs task definitions
+          command: |
+            . venv/bin/activate
+            ./scripts/instantiate.sh
+            (cd ml-papaya-extractor && globality-build build-gen local)
+            eval $(aws ecr get-login --no-include-email)
+            globality-build docker-base --repo ml-papaya-extractor --base-dir ml-papaya-extractor
+            globality-build docker --repo ml-papaya-extractor --base-dir ml-papaya-extractor
```
Comment (pokey) [inline .circleci/config.yml:81]: These two commands build a docker image of the instantiated cookiecutter on which to run tests / lint / mypy.  They leverage the `--base-dir` option added in globality-corp/globality-build#161.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+      - setup_remote_docker
+
+      - run:
+          name: Build Base Docker
+          # install dependencies for loading ecs task definitions
+          command: |
+            . venv/bin/activate
+            ./scripts/instantiate.sh
+            (cd ml-papaya-extractor && globality-build build-gen local)
```
Comment (pokey) [inline .circleci/config.yml:78]: 1. Activate virtualenv to get `cookiecutter` command
2. Instantiate the cookiecutter
3. Run `globality-build` on the instantiated cookiecutter

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/bundles/example.py [hunk 3]
```diff
@@ -2,10 +2,7 @@
 
 from microcosm.api import binding, defaults
 from microcosm_logging.decorators import logger
-from microcosm_sagemaker.artifact import (
-    BundleInputArtifact,
-    BundleOutputArtifact,
-)
+from microcosm_sagemaker.artifact import BundleInputArtifact, BundleOutputArtifact
```
Comment (pokey) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/bundles/example.py:5]: Turned out we had a `flake8` violation!  😎

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+#  next version of the build.
+#
+#  See: http://github.com/globality-corp/globality-build
+#
+#
+
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: ${AWS_ECR_DOMAIN}/globality-build:fe815e7fe622e84fba3ea7c6f72299c4a052f414
```
Comment (pokey) [inline .circleci/config.yml:17]: Using the tag here as temporary workaround until new `globality-build` is released incorporating globality-corp/globality-build#161

**Comment by adamhadani** [comment]
@pokey good to merge or still WIP? let me know if you need any help from me I assume you took over this PR..

**Review by KensoDev — APPROVED** [review]

**Pull request merged**